hcl2template/types.variables.go

```diff
@@ -375,3 +378,13 @@ func (variables Variables) collectVariableValues(env []string, files []*hcl.File
 
 	return diags
 }
+
+func expressionFromVariableDefinition(name string, value string, variableType cty.Type) (hclsyntax.Expression, hcl.Diagnostics) {
+	switch variableType {
+	case cty.String, cty.Number:
+		return &hclsyntax.LiteralValueExpr{Val: cty.StringVal(value)}, nil
+	default:
+		fakeFilename := fmt.Sprintf("<value for var.%s from arguments>", name)
```

Comment:
Here we don't know where the value was read from, and it can be read `from arguments` or `from env`, do you think it would from a good idea to pass the from string here ?

The outcome is nice because then the user has better context on a potential error.

Reply:
Ah yes! good catch. I'll add that back in.